app/controllers/reviews_controller.rb

```diff
@@ -19,12 +16,12 @@ def create
   end
 
   def edit
-    @review = current_user.reviews.find(params[:id])
-    @buddy = Buddy.find(params[:buddy_id])
+    @review = Review.find(params[:id])
+    @buddy = @review.buddy_id
   end
 
   def update
-    @review = current_user.reviews.find(params[:id])
+    @review = Review.find(params[:id])
```

Comment:
Revert this, you want to keep the `current_user` scope to ensure that only the user who wrote the review can update it.